packages/app/background-jobs-common/src/background-job-processor/barrier/ChildJobThrottlingBarrier.ts

```diff
@@ -0,0 +1,37 @@
+import type { AbstractBackgroundJobProcessor } from '../processors/AbstractBackgroundJobProcessor'
+import type { BaseJobPayload } from '../types'
+import type { BarrierCallback } from './barrier'
+
+export type ChildJobThrottlingBarrierConfig = {
+  retryPeriodInMsecs: number
+  maxChildJobsInclusive: number
+}
+
+export type ChildJobThrottlingBarrierContext = {
+  // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+  childJobProcessor: AbstractBackgroundJobProcessor<any>
+}
+
+/**
+ * Limit amount of child jobs that can exist at a time.
+ * Note that for performance reasons it performs an optimistic check and can overflow in highly concurrent systems,
+ * so it is recommended to use lower values for max child jobs, to preserve a buffer for the overflow
+ */
+export function createChildJobThrottlingBarrier(
```

Comment:
🟢 I'm a bit concerned that the term `child` might conflict with the [BullMQ flow's](https://docs.bullmq.io/guide/flows) terminology, potentially causing confusion. Is there a more suitable term we could use instead?
E.g. more abstract `createJobQueueSizeThrottlingBarrier `